esp-hal/src/dma/gdma.rs

```diff
@@ -606,3 +614,123 @@ impl<'d> Dma<'d> {
         }
     }
 }
+
+pub use m2m::*;
+mod m2m {
+    use embedded_dma::{ReadBuffer, WriteBuffer};
+
+    use super::dma_private::{DmaSupport, DmaSupportRx};
+    use crate::dma::{
+        Channel,
+        ChannelTypes,
+        DescriptorChain,
+        DmaDescriptor,
+        DmaError,
+        DmaPeripheral,
+        DmaTransferRx,
+        RxPrivate,
+        TxPrivate,
+    };
+
+    /// DMA Memory to Memory pseudo-Peripheral
+    ///
+    /// This is a pseudo-peripheral that allows for memory to memory transfers.
+    /// It is not a real peripheral, but a way to use the DMA engine for memory
+    /// to memory transfers.
+    pub struct Mem2Mem<'d, C, MODE>
+    where
+        C: ChannelTypes,
+        MODE: crate::Mode,
+    {
+        channel: Channel<'d, C, MODE>,
+        tx_chain: DescriptorChain,
+        rx_chain: DescriptorChain,
+        peripheral: DmaPeripheral,
+    }
+
+    impl<'d, C, MODE> Mem2Mem<'d, C, MODE>
+    where
+        C: ChannelTypes,
+        MODE: crate::Mode,
+    {
+        /// Create a new Mem2Mem instance.
+        ///
+        /// # Safety
+        ///
+        /// You must insure that your not using DMA for the same peripheral and
+        /// that your the only one using the DmaPeripheral.
+        pub unsafe fn new(
```

Comment:
I'd love to have a safe constructor as well as this unsafe one. Maybe tag all "DMA" available peripherals with some trait (Maybe one that returns the DmaPeripheral value)  Then the safe constructor could take the peripheral and that would insure its DMA is not already used. More ideas?

Reply:
> Maybe tag all "DMA" available peripherals with some trait (Maybe one that returns the DmaPeripheral value)

Sounds like a solution. We probably should introduce dummy peripherals for Mem2Mem on C6 and later 🤔

Reply:
@bjoernQ I called them `Mem2MemN` - maybe I should have used `DummyN`. Should change them?